app/src/main/java/com/karumi/data/repository/NetworkSuperHeroDataSource.kt

```diff
@@ -0,0 +1,65 @@
+package com.karumi.data.repository
+
+import com.karumi.domain.model.AuthDomainError
+import com.karumi.domain.model.DomainError
+import com.karumi.domain.model.NotInternetDomainError
+import com.karumi.domain.model.SuperHero
+import com.karumi.domain.model.UnknownDomainError
+import com.karumi.marvelapiclient.CharacterApiClient
+import com.karumi.marvelapiclient.MarvelAuthApiException
+import com.karumi.marvelapiclient.model.CharacterDto
+import com.karumi.marvelapiclient.model.MarvelImage
+import com.karumi.marvelapiclient.model.MarvelResponse
+import org.funktionale.either.Either
+import java.net.ConnectException
+
+class NetworkSuperHeroDataSource(val apiClient: CharacterApiClient) : SuperHeroDataSource {
+    companion object {
+        val PAGE_SIZE = 10
+        val RESPONSE_OK = 200
+    }
+
+    override fun get(key: String): Either<DomainError, SuperHero> =
+        try {
+            mapResponse(apiClient.getCharacter(key), { mapSuperHero(it) })
+        } catch (exception: Exception) {
+            Either.left(mapException(exception))
+        }
+
+    override fun isUpdated(): Boolean = true
```

Comment:
If `contains` already have a default value, I'd say that `isUpdated` should have a default value as well.

Reply:
true, and in contains method I can remove the brackets